---
title: "Theory"
output: html_document
runtime: shiny
---
<style type="text/css">

h1.title {

text-align: center;
color: DarkBlue;
font-size: 38px;

}

p{

font-size: 18pt;
font-family: times, serif;
}

p1{

font-size: 14pt;
font-family: times, serif;
}

</style>

<p>
The ordinary differential equations (ODE’s) we use in GeneEx are chemical rate equations that model the expression of a gene (node) in a GRN. We then use numerical methods to solve these ODE’s and output the steady state gene expression of each gene in the GRN. Each equation takes on the following general form:
</p>

<p>
$$ 
Rate \:of\:gene\: expression = Production\: rate - Degradation\: rate
$$
</p>

<p>
In the most elementary example, the dynamics of an isolated gene ($B$) is modeled using the differential equation:
</p>

<p>
$$
\frac{d[B_t]}{dt} = g_B - k_B[B_t],
$$
</p>

<p>
where $B_t$ is defined as the expression of gene $B$ at time $t$, and $g_B$ and $k_B$, are parameters that represent the genes basal production and degradation rates, respectively. In this straightforward case, the steady state is the expression level of gene $B$ when $\frac{d[B_t]}{dt}=0$:
</p>

<p>
$$
B_{t=\infty} = \frac{g_B}{k_B}
$$
</p>

---

<p>
If gene $B$ is regulated by another gene $A$, then their interaction is modeled using the shifted Hill function, an equation that simulates the chemical kinetics of ligand bonding, which we will now briefly derive using a number of fundamental principles in chemistry. Given an activator ($A$), the gene product of gene A, which in order to initative transcription, must bind $n$ times to a promotor site on the DNA of gene B ($P_{BA}$); we can use the principle of conservation of mass to state:
</p>


<p>
$$[P_{BA}] + [A_nP_{BA}] = [P_T], $$
</p>

<p>
where $P_T$ is the total concentration of bound and unbound promotor. We are neglecting intermediate states where fewer than $n$ activators are bound. Subsequently, using the theory of mass-action kinetics, we can state that the rate of concentration change for the activator-promotor complex is:
</p>

<p>
$$ \frac{d[A_nP_{BA}]}{d{t}} = k_{on}[A]^n[P_{BA}] - k_{off}[A_nP_{BA}],$$
</p>

<p>
where $k_{on}$ and $k_o{ff}$ are the association and dissociation rate constants, respectively. Using the steady-state solution at which  $\frac{d[A_nP_{BA}]}{d{t}} = 0$, as well as the conservation of mass equation mentiond earlier, we can obtain **the fraction of unbound promotor**, also known as the Hill-Langmuir function:
</p>

<p>
$$ \frac{[P_{BA}]}{P_T} = \frac{1}{1+(\frac{[A]}{T_{BA}})^n} = H(A, T_{BA}, n), $$
</p>

<p>
where $n$ is the Hill coefficient and $T_{BA}$ is the threshold constant and is defined as:
</p>

<p>
$$T_{BA} = \sqrt[n]{\frac{k_{off}}{k_{on}}}$$
</p>

<p>
Although we derived the equations by assuming that $n$ is the number of binding sites, $n$ is commonly defined as the cooperativity of the regulation in the Hill function, and will be unique for every interaction ($n_{BA}$, in the example above). 
</p>


<p>
In the simple case where gene $B$ is activated by gene $A$, we can use a variation of the Hill function, known as the shifted Hill function($H_s$), to model **the production rate** of gene B:
</p>

<p>


</p>

<p>

$$
\begin{aligned}
 Production \:rate \:of \:B &= Production \:of\: B\: when \:A \:is \:NOT \:bound + Production\: of\: B\: when\: A\: IS\: bound \\
  &= g_{B}[H(A, T_{BA}, n) + \lambda^+_{BA}(1- H(A, T_{BA}, n))] \\
  &= g_{B}(\lambda^+_{BA} + \frac{1-\lambda^+_{BA}}{{1+(\frac{[A]}{T_{BA}})^n}}) \\
  &= g_{B}H_{s}(A, T_{BA}, n_{BA}, \lambda^+_{BA}),  
\end{aligned}
$$
</p>


<p>
where $λ_{BA}^+$ is the fold change of the regulation, or in other words, the degree to which the basal production rate is magnified, in the case of an activator, or diminished, in the case of an inhibitor ($λ_{BA}^-$). 
</p>

<p>
Using the shifted Hill function, we can now construct the ODE that models the rate of expression change of gene B at time t ($B_t$) as a function of activator A at time t ($A_t$):
</p>

<p>
$$\frac{dB_t}{d{t}} =  g_{B}H_{s}(A_t, T_{BA}, n_{BA}, \lambda^+_{BA}) -k_BB_t$$
</p>

<p>
The various tools in GeneEx implement a slightly modified version of the above rate equation that put the basal production $g_B$ in terms of the **maximal** production, $G_B$, where:
</p>

<p>
$$g_B = G_B/\lambda^+_{BA}  \: \: (for \: an \: activator )\\
g_B = G_B  \: \:  (for \: an \: inhibitor )$$
</p>

<p>
This variation is more biologically relevant because gene expression has an upper limit in living systems. 
</p>

<p>
For the more typical case where gene $B$ has multiple regulators, the production parameter is multiplied by the shifted Hill function of each regulator. Here we can see the ODE for the rate of expression change of gene B at time t, when regulated by activator A and inhibitor C:
</p>

<p>
$$\frac{dB_t}{d{t}} =  G_{B}*\frac{H_{s}(A_t, T_{BA}, n_{BA}, \lambda^+_{BA})}{\lambda^+_{BA}}*H_{s}(C_t, T_{BC}, n_{BC}, \lambda^-_{BC}) -k_BB_t$$
</p>

---

PARAGRAPH ABOUT NOISE AND INCLUDE NOISE EQUATION
</p>

<p>
$$\frac{dB_t}{d{t}} =  g_{B}*H_{s}(P_{BA}, T_{BA}, n_{BA}, \lambda^+_{BA})*H_{s}(P_{BC}, T_{BC}, n_{BC}, \lambda^-_{BC}) -k_BB_t$$
</p>


<p>
The ODE’s built in GeneEx are solved using variations of the Euler method including Heun’s method and the Runge Kutta method. The steady state expression value of gene $B$ can be numerically approximated using the following general form of the Euler method: 
</p>

<p>
$$B_{t+1} = B_t + \frac{dB_t}{dt}\Delta t,$$
</p>

<p>
where $\Delta t$ is the step size. The simulation time, which is the amount of steps, as well as the step size, can both be specified by the user in GeneEx to regulate the speed and accuracy of the steady approximations for each gene in the network. 
</p>
